mso/provider_test.go

```diff
@@ -156,3 +163,64 @@ func customTestCheckResourceTypeSetAttr(resourceName, resourceAttrRootkey string
 		return nil
 	}
 }
+
+func setupTestLogCapture(t *testing.T, logLevel string) string {
+	logFile, err := os.CreateTemp("", "tf-acc-test-*.log")
+	if err != nil {
+		t.Fatalf("Failed to create temp log file: %v", err)
+	}
+
+	t.Cleanup(func() {
+		logFile.Close()
+		os.Remove(logFile.Name())
```

Comment:
should we store logFile.Name() as variable and reuse, since it is called 3 times?